src/swish/io.ss

```diff
@@ -349,9 +367,7 @@
   ;; Console Ports
 
   (define (make-console-input)
-    ;; no need to register static stdin port with the guardian
-    (define name "stdin-nb")
-    (binary->utf8 (make-iport name (make-osi-port name (erlang:now) (osi_get_stdin)) #f)))
+    (binary->utf8 (make-osi-input-port (open-fd-port "stdin-nb" 0 #f))))
 
   (define hook-console-input
```

Comment:
Because Swish uses stderr as the mode of last resort when reporting problems, I would prefer not to hook it. I think that hooking stdout might be OK, but it's already dangerous to write to stdout from multiple processes at the same time.